src/lib.rs

```diff
@@ -680,26 +787,44 @@ fn encode_partition(fi: &FrameInvariants, fs: &mut FrameState, cw: &mut ContextW
 
     match partition {
         PartitionType::PARTITION_NONE => {
-            // TODO(anyone): Until we have RDO-based block size decision,
-            // call rdo_mode_decision() for a partition.
-            let rdo_none = rdo_mode_decision(fi, fs, cw, bsize, bo);
-            cw.bc.set_mode(bo, bsize, rdo_none.pred_mode);
+            let pred_mode = if rdo_output.part_modes.len() > 0 { 
+                rdo_output.part_modes[0].pred_mode
+            } else { // edges
+                rdo_mode_decision(fi, fs, cw, bsize, bo).part_modes[0].pred_mode };
```

Comment:
As a record or comments for readers, this block of no-split partition comes from the case which does not go through rdo_partition_decision(). For ex, unconditionally split 32x32 partition, or all unconditional 4x4s on right/bottom frame border.

Reply:
I have added more detailed comments in this function to clarify how the partitioning decisions are made.